roles/kafka_controller/tasks/health_check.yml

```diff
@@ -1,26 +1,90 @@
 ---
 # health check for kafka controller
-- name: Check Kafka Metadata Quorum
+- name: Check Kafka Metadata Quorum using bootstrap controller
   shell: |
     {{ binary_base_path }}/bin/kafka-metadata-quorum --bootstrap-controller {{inventory_hostname}}:{{kafka_controller_port}} \
       --command-config {{kafka_controller.client_config_file}} describe --replication
   environment:
     KAFKA_OPTS: "-Xlog:all=error -XX:+IgnoreUnrecognizedVMOptions {% if kerberos_client_config_file_dest != '/etc/krb5.conf' %}-Djava.security.krb5.conf={{kerberos_client_config_file_dest}}{% endif %}"
-  ignore_errors: false
+  ignore_errors: true
   changed_when: false
   check_mode: false
+  register: result1
 
 #Registers LEO of controllers only if Metadata Quorum passed in the above task
-- name: Register LogEndOffset
+- name: Register LogEndOffset using bootstrap controller
   shell: |
     {{ binary_base_path }}/bin/kafka-metadata-quorum --bootstrap-controller {{inventory_hostname}}:{{kafka_controller_port}} \
       --command-config {{kafka_controller.client_config_file}} describe --replication |  grep -v Observer | awk '{print $2}'
   environment:
     KAFKA_OPTS: "-Xlog:all=error -XX:+IgnoreUnrecognizedVMOptions {% if kerberos_client_config_file_dest != '/etc/krb5.conf' %}-Djava.security.krb5.conf={{kerberos_client_config_file_dest}}{% endif %}"
   register: leo
+  changed_when: false
+  check_mode: false
+  when: result1.rc == 0
+
+- name: Check LogEndOffset values
+  assert:
+    that:
+      - "{{ item|int > 0 and leo.stdout_lines[1:]|max|int - item|int < 1000 }}"
+    fail_msg: "UnreachableQuorumMember or Found at least one quorum voter with an offset {{ item }}, while the primary controller was at offset {{ leo.stdout_lines[1:]|max}}
+               The max allowed offset lag is 1000"
+  loop: "{{ leo.stdout_lines[1:] }}"
+  ignore_errors: false
+  changed_when: false
+  check_mode: false
+  when: result1.rc == 0
+
+- name: Add confluent.use.controller.listener config to Client Properties
```

Comment:
we should move this to template, similar to 7.6.x

Reply:
But it was getting added each time even when not needed. Now this property is only getting added when needed. so is there any benefit to putting it in template

Reply:
yes I understand that, I think it's just cleaner and will be uniform across branches that way. As long as the config just lying there does not cause failure, I think we're good but if the support for the config also has been dropped then we'll need to do it this way